### PR TITLE
feat: Customize node groups and user-data

### PR DESCRIPTION
**Pull request by ichaban-pinterest**
## Summary
Added several customizations for:
- `user-data` module and templates: added support for `format_mount_nvme_disk`, `kubelet_extra_args` for Linux template only
- Managed Node Groups: added support for `format_mount_nvme_disk`, `kubelet_extra_args`; removed `ignore_changes` for `desired_size` from scaling config
- Self-Managed Node Groups: added support for `kubelet_extra_args`, `labels`

### CODE REVIEW

**Review by ashrestha-pinterest — APPROVED** [review]

File: templates/linux_user_data.tpl [hunk 1]
```diff
@@ -1,12 +1,48 @@
...
+mount -a
+%{ endif ~}
+%{ if length(cluster_service_ipv4_cidr) > 0 ~}
+export SERVICE_IPV4_CIDR=${cluster_service_ipv4_cidr}
+%{ endif ~}
 %{ if enable_bootstrap_user_data ~}
 B64_CLUSTER_CA=${cluster_auth_base64}
 API_SERVER_URL=${cluster_endpoint}
-/etc/eks/bootstrap.sh ${cluster_name} ${bootstrap_extra_args} --b64-cluster-ca $B64_CLUSTER_CA --apiserver-endpoint $API_SERVER_URL \
-  --ip-family ${cluster_ip_family} --service-${cluster_ip_family}-cidr ${cluster_service_cidr}
```
Comment (sfbahr) [inline templates/linux_user_data.tpl:10]: Are we sure we want to skip these additions to the bootstrap.sh script?

Comment (sfbahr) [reply templates/linux_user_data.tpl:10]: They did not exist on the old fork, I think we should be keeping these last two arguments: https://github.com/pinterest/terraform-aws-eks/pull/3/files#diff-5e40a572eaa3c9cddd3c65bbbff6fc1477976db1b37e3b56e4d2aaa914b94eabL12

**Review by sfbahr — CHANGES_REQUESTED** [review]

File: templates/linux_user_data.tpl (hunk 1 above)
Comment (sfbahr) [reply templates/linux_user_data.tpl:10]: I take it back, looks like both of these are inferred from the describe cluster query so our arguments should be fine: https://github.com/awslabs/amazon-eks-ami/blob/2b5acb97f4c6d1971cb7d645e1fdb13b92cee10e/templates/al2/runtime/bootstrap.sh#L387-L391

**Review by sfbahr — APPROVED** [review]
nvm, this looks good

**Pull request merged**